nodejs/aws-infra/experimental/cluster.ts

```diff
@@ -0,0 +1,132 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as mod from ".";
+
+import { Network } from "./../network";
+
+import * as utils from "../utils";
+
+/**
+ * A Cluster is a general purpose ECS cluster configured to run in a provided Network.
+ */
+export class Cluster extends pulumi.ComponentResource {
+    public readonly instance: aws.ecs.Cluster;
```

Comment:
Decision 1.  When wrapping a same-named aws resource, i've called the wrapped type 'instance'.  This is to prevent ugliness like `cluster.cluster...`.  i am not wedded to this decision.

Reply:
:+1:

Reply:
Although I like avoiding `cluster.cluster` (which I agree is ugly), calling it `instance` does have the downside of potentially misleading people into thinking this returns the EC2 instance(s) underlying the cluster.

Reply:
I'm hoping that the 'type' of the variable dissuades people from that idea.  do you have an alternate idea here?

Reply:
todo: brainstorm naming on this.  ideas: "raw" "underlyingInstance".

Reply:
I had thought of `raw` also.

Just to check, are we applying this to _all_ resources? E.g., no more `bucket.bucket`?